tests/server/test_env_settings.py

```diff
@@ -0,0 +1,122 @@
+
+    Contact: code@inmanta.com
+"""
+import pytest
+
+from inmanta import data
+
+
+@pytest.mark.asyncio
+async def test_environment_settings(client, server, environment):
+    """
+        Test environment settings
+    """
+    result = await client.list_settings(tid=environment)
+    assert result.code == 200
+    assert "settings" in result.result
+    assert "metadata" in result.result
+    assert "auto_deploy" in result.result["metadata"]
+    assert len(result.result["settings"]) == 0
+
+    # set invalid value
+    result = await client.set_setting(tid=environment, id="auto_deploy", value="test")
+    assert result.code == 500
+
+    # set non existing setting
+    result = await client.set_setting(tid=environment, id="auto_deploy_non", value=False)
+    assert result.code == 404
+
+    result = await client.set_setting(tid=environment, id="auto_deploy", value=False)
+    assert result.code == 200
+
+    result = await client.list_settings(tid=environment)
+    assert result.code == 200
+    assert len(result.result["settings"]) == 1
+
+    result = await client.get_setting(tid=environment, id="auto_deploy")
+    assert result.code == 200
+    assert not result.result["value"]
+
+    result = await client.get_setting(tid=environment, id="test2")
+    assert result.code == 404
+
+    result = await client.set_setting(tid=environment, id="auto_deploy", value=True)
+    assert result.code == 200
+
+    result = await client.get_setting(tid=environment, id="auto_deploy")
+    assert result.code == 200
+    assert result.result["value"]
+
+    result = await client.delete_setting(tid=environment, id="test2")
+    assert result.code == 404
+
+    result = await client.delete_setting(tid=environment, id="auto_deploy")
+    assert result.code == 200
+
+    result = await client.list_settings(tid=environment)
+    assert result.code == 200
+    assert "settings" in result.result
+    assert len(result.result["settings"]) == 1
+
+    result = await client.set_setting(tid=environment, id=data.AUTOSTART_AGENT_DEPLOY_SPLAY_TIME, value=20)
+    assert result.code == 200
+
+    result = await client.set_setting(tid=environment, id=data.AUTOSTART_AGENT_DEPLOY_SPLAY_TIME, value="30")
+    assert result.code == 200
+
+    result = await client.get_setting(tid=environment, id=data.AUTOSTART_AGENT_DEPLOY_SPLAY_TIME)
+    assert result.code == 200
+    assert result.result["value"] == 30
+
+    result = await client.delete_setting(tid=environment, id=data.AUTOSTART_AGENT_DEPLOY_SPLAY_TIME)
+    assert result.code == 200
+
+    result = await client.set_setting(
+        tid=environment, id=data.AUTOSTART_AGENT_MAP, value={"agent1": "", "agent2": "localhost", "agent3": "user@agent3"}
+    )
+    assert result.code == 200
+
+    result = await client.set_setting(tid=environment, id=data.AUTOSTART_AGENT_MAP, value="")
+    assert result.code == 500
+
+
+@pytest.mark.asyncio
+async def test_environment_settings_v2(client_latest, server, environment):
```

Comment:
vx actually, latest is v2 now, but if the fixture is intended to stay at v2, it should be called v2

Reply:
it is a test, not a fixture. v2 seems most descriptive now.

Reply:
done